### ISSUE TITLE
[spec] Phase A -- honesty foundation: freeze RACE config, blind split, L5 invariant

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/t27#1001]
## Goal

Establish the honesty foundation for the phi->IGLA epic (#181):

1. **A1 (freeze+seal)** -- introduce `specs/ml/optimizer/race_config.t27` that
   freezes the RACE optimizer config under a content hash (SHA-256 of canonical
   TOML) and is sealed via `./scripts/tri seal`. Includes a Chinchilla ~20
   tok/param budget note (Hoffmann 2022, arXiv 2203.15556).

2. **A2 (80/20 blind split)** -- record the held-out eval-split protocol in
   `race_config.t27` (split seed = 17, 80% train / 20% held-out) so champion
   BPB=2.1919 can be re-baselined on a shard not seen during training or ASHA
   pruning.

3. **A3 (Trinity runtime invariant)** -- add L5 identity blocks to
   `specs/numeric/phi_ratio.t27`: `phi^2 + phi^-2 = 3` [Verified, the ONLY
   [Verified] phi-fact] and `phi^2 = phi + 1` [Verified]. f64 tolerance 1e-14.
   Re-seal after edit.

All phi optimizer claims remain **[Open conjecture]** until the control test
(Phase B1) decides. `delta_CP = 3/phi^2` is **[Retracted]** and must not be
cited.

Parent: #181

## Why this matters

Without a frozen config, BPB numbers cannot be reproduced or attributed to a
specific hyperparameter set. Without a held-out split, BPB=2.1919 may be an
in-sample artefact. Without the L5 invariant, the GOLDEN CHAIN has no runtime
guard that the foundational phi constant is correct to f64 precision. These
three items are pre-conditions for honest phi-specificity claims.

Falsification-first: if a non-phi control of equal tuning budget matches phi
within Monte-Carlo error, phi-specificity is FALSIFIED. This must be stated
plainly in every result table derived from the RACE.

## Source of truth

- New spec (A1 + A2): `specs/ml/optimizer/race_config.t27`
- Patched spec (A3):  `specs/numeric/phi_ratio.t27`
- Seal location A1:   `.trinity/seals/RaceConfig.json`
- Seal location A3:   `.trinity/seals/PhiRatio.json` (updated)
- Reference trainer:  `gHashTag/trios-trainer-igla` `src/optimizer.rs`
  (phi-canonical anchors: beta1=phi^-1, wd=phi^-3, lr=phi^-3)

## Deliverable

1. `specs/ml/optimizer/race_config.t27` -- new module `RaceConfig` with:
   - Constants: phi-canonical anchors (RACE_BETA1_PHI_CANONICAL, etc.)
   - Constants: champion config (hidden=828, steps=81000, seed=43)
   - Constants: split protocol (SPLIT_SEED=17, TRAIN_FRACTION=0.80)
   - Chinchilla budget note referencing Hoffmann 2022
   - Four-arm control test design (phi-canonical, phi-damped, tuned-standard,
     random-rational)
   - `test`, `invariant`, `bench` blocks (L4 compliance)

2. `specs/numeric/phi_ratio.t27` (patch) -- new blocks inside module PhiRatio:
   - `test test_l5_phi_sq_equals_phi_plus_one`
   - `test test_l5_phi_sq_plus_phi_inv_sq_equals_3`
   - `test test_l5_phi_inv2_equals_2_minus_phi`
   - `invariant l5_phi_sq_equals_phi_plus_one`
   - `invariant l5_phi_sq_plus_phi_inv_sq_equals_3`
   - `bench bench_l5_phi_identity_check`

3. Updated seals:
   - `.trinity/seals/RaceConfig.json` (new)
   - `.trinity/seals/PhiRatio.json` (updated spec_hash)

## Done when

- `./scripts/tri test` passes for both `RaceConfig` and `PhiRatio` modules.
- `./scripts/tri validate-conformance` passes (FORMAT-SPEC-001.json intact).
- `.trinity/seals/RaceConfig.json` exists with valid `spec_hash`.
- `.trinity/seals/PhiRatio.json` has updated `spec_hash` matching current file.
- `./scripts/tri validate-gen-headers` passes (gen/ not hand-edited, L2).
- `docs/NOW.md` and root `NOW.md` updated to today's date (`tri check-now`).
- All phi optimizer claims in the new spec carry `[Open conjecture]` labels.
- `phi^2 + phi^-2 = 3` is labelled `[Verified]` and described as the ONLY
  [Verified] phi-fact.

## How to verify

```bash
# From the t27 repo root (/tmp/t27 in the dev environment):

# 1. Verify the new spec parses and tests pass
./scripts/tri test specs/ml/optimizer/race_config.t27
./scripts/tri test specs/numeric/phi_ratio.t27

# 2. Seal both specs
./scripts/tri seal specs/ml/optimizer/race_config.t27 --save
./scripts/tri seal specs/numeric/phi_ratio.t27 --save

# 3. Confirm seals written
ls -la .trinity/seals/RaceConfig.json
ls -la .trinity/seals/PhiRatio.json

# 4. Validate conformance (must not break FORMAT-SPEC-001.json / gf16.t27, L6)
./scripts/tri validate-conformance

# 5. Validate gen headers (L2 -- gen/ must not be hand-edited)
./scripts/tri validate-gen-headers

# 6. NOW.md gate (L1)
./scripts/tri check-now

# 7. Run full suite to confirm no regressions
./scripts/tri test
```

## Risks

- `RACE_BETA1_PHI_DAMPED` constant has trailing ellipsis in draft (0.9/PHI is
  irrational; exact decimal must be resolved to f64 hex or enough digits).
  Parent agent: fill in `0.5559321608040201` (0.9 / 1.618033988749895).

- `CORPUS_SHA256` is TBD; parent agent must compute SHA-256 of
  `fineweb_train.bin` before sealing.

- `CHAMPION_BPB_HELD_OUT` is TBD; re-baseline run required.

- `CONFIG_HASH_TBD` requires normalising the TOML and computing SHA-256 of
  the canonical text. Parent agent: implement `tri freeze-config` or do this
  manually and hard-code the result.

- `seal-coverage.yml` CI may require a new entry for `RaceConfig`. Check
  `.github/workflows/seal-coverage.yml` before pushing.

- `phi_ratio.t27` already has a PhiRatio seal (`PhiRatio.json` and
  `numeric_PhiRatio.json` -- two seal files for the same spec path). Parent
  agent should confirm which seal file CI actually checks and update the right
  one.

- The spec uses `sacred_physics::PHI` but the module currently imports
  `math::constants` and `math::sacred_physics`. Verify the import path is
  consistent with how the existing phi_ratio.t27 already references it.

## Links

- Parent epic: #181 (phi->IGLA)
- Hoffmann 2022 (Chinchilla): https://arxiv.org/abs/2203.15556
- Li & Talwalkar 2019 (random search): https://arxiv.org/abs/1902.07638
- Melis/Dyer/Blunsom 2017 (equal-budget): https://arxiv.org/abs/1707.05589
- Xie & Li 2024 (AdamW l-inf): https://arxiv.org/abs/2404.04454
- Wang & Aitchison 2025 (wd scaling): https://arxiv.org/abs/2405.13698
